client/scripts/DrawTogether.js

```diff
@@ -1609,18 +1718,37 @@ DrawTogether.prototype.createFavorite = function (x, y, name) {
 };
 
 DrawTogether.prototype.createProtectedRegion = function (from, to) {
-	if (!this.memberlevel) {
-		this.chat.addMessage("Creating a protected region is a member only feature.");
+	if (typeof this.reputation === 'undefined') { 
+		this.chat.addMessage("You must be logged in to create protected regions.");
 		return;
 	}
+	var regionCount = this.regionsList.length || 0;
+
+	if (!this.memberlevel) {
+		if(this.reputation < this.REGION_MIN_REP){
+			
+			this.chat.addMessage("You must have at least "+ this.REGION_MIN_REP +" rep!");
+			return;
+		}
+		if(typeof this.regionsList !== 'undefined')
+		{
+			if(typeof this.regionsList.length === 'number')
+				if(this.regionsList.length > 1)
+				{
+					this.chat.addMessage("Having more than one region is premium only!");
+					return;
+				}
+		}
+	}
 
-	this.network.socket.emit("createprotectedregion", from, to, function (err, result) {
+	this.network.socket.emit("createprotectedregion", from, to, regionCount, function (err, result) {
```

Comment:
This is not safe. This does not fix the fact that the amount of regions is checked client side. So we are still trusting the client. If you want to do the premium check server side then getting the amount of regions is also something you have to do server side.

Preferably on the playerserver.